WalletWasabi/Services/UpdateManager.cs

```diff
@@ -0,0 +1,243 @@
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+using WalletWasabi.Tor.Http;
+
+namespace WalletWasabi.Services;
+
+public class UpdateManager
+{
+	private string InstallerName { get; set; } = "";
+	private const byte MaxTries = 3;
+	private const string ReleaseURL = "https://api.github.com/repos/zkSNACKs/WalletWasabi/releases/latest";
+
+	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient)
+	{
+		DataDir = dataDir;
+		DownloadsDir = Path.Combine(DataDir, "Downloads");
```

Comment:
We don't need the `DataDir` property at all IMO.  We only use DataDir at places where `DownloadsDir` could be used, too.

So I think it's redundant to save it to a property.

```suggestion
		DownloadsDir = Path.Combine(dataDir, "Downloads");
```